src/evergreen/api.py

```diff
@@ -212,10 +212,17 @@ def _paginate(
         """
         response = self._call_api(url, params)
         json_data = response.json()
+
         while "next" in response.links:
-            if params and "limit" in params and len(json_data) >= params["limit"]:
-                break
-            response = self._call_api(response.links["next"]["url"])
+            if params and "limit" in params:
+                if len(json_data) >= params["limit"]:
+                    # the limit has been reached - return
+                    break
+                else:
+                    params["limit"] -= len(json_data)
```

Comment:
What case is this trying to handle? 

I think this will prevent the `_paginate` function from reaching the full amount of desired results as it's moving the given limit if I'm reading this correctly.

Reply:
yeah, I think you are right - thank you